lib/Sema/TypeCheckDecl.cpp

```diff
@@ -3430,19 +3433,25 @@ static void validateTypealiasType(TypeChecker &tc, TypeAliasDecl *typeAlias) {
 }
 
 
-/// Bind the given function declaration, which declares an operator, to
-/// the corresponding operator declaration.
-void bindFuncDeclToOperator(TypeChecker &TC, FuncDecl *FD) {
-  OperatorDecl *op = nullptr;
+/// Bind the given function declaration, which declares an operator, to the corresponding operator declaration.
+llvm::Expected<OperatorDecl *>
+FunctionOperatorRequest::evaluate(Evaluator &evaluator, FuncDecl *FD) const {
+  // Fast-path: Most functions are not operators.
+  if (!FD->isOperator()) {
```

Comment:
Move this to FuncDecl::getOperator(), so that we avoid populating the cache with a null value in this case; then replace the check here with an assert().